encodings/decimal/src/decimal_wrapper/mod.rs

```diff
@@ -0,0 +1,159 @@
+mod serde;
+
+use vortex_array::arrays::DecimalArray;
+use vortex_array::stats::{ArrayStats, StatsSetRef};
+use vortex_array::variants::{DecimalArrayTrait, PrimitiveArrayTrait};
+use vortex_array::vtable::{ComputeVTable, VTableRef};
+use vortex_array::{
+    Array, ArrayCanonicalImpl, ArrayImpl, ArrayOperationsImpl, ArrayRef, ArrayStatisticsImpl,
+    ArrayValidityImpl, ArrayVariants, ArrayVariantsImpl, Canonical, Encoding, ProstMetadata,
+    try_from_array_ref,
+};
+use vortex_dtype::{DType, DecimalDType, match_each_signed_integer_ptype};
+use vortex_error::{VortexExpect, VortexResult, vortex_bail};
+use vortex_mask::Mask;
+use vortex_scalar::{DecimalValue, Scalar};
+
+use crate::decimal_wrapper::serde::DecimalWrapperMetadata;
+
+#[derive(Clone, Debug)]
+pub struct DecimalWrapperArray {
```

Comment:
What's this array actually for? Seems like it sort of performs decimal narrowing, but then also stores the decimal internally as an integer type? 

Decimal narrowing should be achievable without a special array, similar to integer narrowing. 
And as for a compressed form, I'm guessing we'd see some benefit from pulling apart the components of a decimal into separate children such that bit-packing is benefitial?

Reply:
The reason I choose this approach are:

 1. Some decimals cannot be represented (for now) and NativePTypes (Primitives), i128+  
 2. I am trying to mirror the canonical primitive array (array without children)
 3. Simpler to add

Reply:
Decimal having children would make this canonical type unusual, the only non-container with children

Reply:
Decimal having children would make this canonical type unusual, the only non-container with children